frontend/src/services/api.js

```diff
@@ -50,20 +50,50 @@ export async function fetchUserById(userId) {
   }
 }
 
+export async function fetchAllStocks() {
```

Comment:
```suggestion
export function fetchAllStocks() {
```

There's no asynchronous operations in this function so you could remove the async here and the awaits that you have before calling it. It would make this function call run synchronously

Reply:
you could also potentially remove the async from the parent functions